Update GHA actions in CI pipeline

AAriam opened the pull request:
## Description

This is a minor update to the CI workflow to make it run again with the current configurations. Ready to merge into main.

AndreaVolkamer reviewed (commented):
@AAriam thanks for looking into this. just wondering why all tests still fail?

AAriam commented:
@AndreaVolkamer Before this PR the CI was failing due to outdated actions and workflow configurations, so the tests were not even executed. To keep commits clear, this PR only addresses those issues in the CI workflow. Now the workflow runs, but fails due to other reasons, namely outdated dependencies in OpenCADD and problems in the superposition subpackage. Some of these issues are addressed in #157. So the idea is that we squash-merge this PR into main, then update #157 from main. If you approve the PR I can do the rest.

AndreaVolkamer reviewed (approved):
thanks!
